common/CMakeLists.txt

```diff
@@ -1,31 +1,2 @@
-set(UNIFIED_SOURCES
-    base/device_matrix_data_kernels.cpp
-    base/index_set_kernels.cpp
-    components/absolute_array_kernels.cpp
-    components/fill_array_kernels.cpp
-    components/format_conversion_kernels.cpp
-    components/precision_conversion_kernels.cpp
-    components/reduce_array_kernels.cpp
-    distributed/partition_kernels.cpp
-    matrix/coo_kernels.cpp
-    matrix/csr_kernels.cpp
-    matrix/dense_kernels.cpp
-    matrix/ell_kernels.cpp
-    matrix/hybrid_kernels.cpp
-    matrix/sellp_kernels.cpp
-    matrix/sparsity_csr_kernels.cpp
-    matrix/diagonal_kernels.cpp
-    multigrid/pgm_kernels.cpp
-    preconditioner/jacobi_kernels.cpp
-    solver/bicg_kernels.cpp
-    solver/bicgstab_kernels.cpp
-    solver/cg_kernels.cpp
-    solver/cgs_kernels.cpp
-    solver/common_gmres_kernels.cpp
-    solver/fcg_kernels.cpp
-    solver/gcr_kernels.cpp
-    solver/gmres_kernels.cpp
-    solver/ir_kernels.cpp
-    )
-list(TRANSFORM UNIFIED_SOURCES PREPEND ${CMAKE_CURRENT_SOURCE_DIR}/unified/)
-set(GKO_UNIFIED_COMMON_SOURCES ${UNIFIED_SOURCES} PARENT_SCOPE)
+add_subdirectory(unified)
+set(GKO_UNIFIED_COMMON_SOURCES ${GKO_UNIFIED_COMMON_SOURCES} PARENT_SCOPE)
```

Comment:
do you need to set that to `PARENT_SCOPE` again? it is set in the common/unified/CMakeLists.txt

Reply:
yes, we need to propagate it up two levels. BTW, this change is not strictly necessary, I removed the code that required it.